docs(content-releases): add missing endpoints to backend docs (#19302)
strapi/strapi: docs/docs/docs/01-core/admin/05-features/hooks/use-document.mdx

---
title: unstable_useDocument
description: API reference for the useDocument hook
tags:
  - admin
  - hooks
  - document
  - validations
---

:::caution
This hook is not stable and **likely to change**. Use at your own risk.
:::

A hook that returns utilities to work with documents.

## Usage

```
function Component() {
  const { validate } = unstable_useDocument();
}
```

At the moment, useDocument only returns a validate function.

### `validate()`

```ts
const validate(
  entry: Entity,
  {
    contentType: Schema.ContentType,
    components: {
      [key: Schema.Component['uid']]: Schema.Component;
    },
    isCreatingEntry?: boolean;
  }
): {
  errors: { [key: string]: TranslationMessage }
}
```

With this function, you can apply the same validations we use in the admin. Please be aware that for this function to work, you need to ensure the following:

> 1. `entry` is a fully populated entry. If you are validating a Content Type with required fields that are relations/components/dynamic zones, you need to send these fields populated in the object. Otherwise, the validation will fail.

Reviewer: ```suggestion
1. If you are validating an entry with relations/components/dynamic zones then it **MUST** be populated. Otherwise, the validation will fail.
```

Reviewer: If the entry does not populate will the validation fail? I think it will pass right? Isn't that the issue?
